### PR TITLE
[stable/redis] Make volumePermissions init container optional

### PR DESCRIPTION
**Pull request by javsalgar**
Signed-off-by: Javier J. Salmeron Garcia <jsalmeron@bitnami.com>

#### What this PR does / why we need it:
The default volume permissions init container does not work in systems like Openshift. This was created for special cases such as cephfs (https://github.com/helm/charts/pull/7928). This PR makes that init container optional. 
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/redis/README.md [hunk 1]
```diff
@@ -167,6 +167,7 @@ The following table lists the configurable parameters of the Redis chart and the
 | `master.readinessProbe.successThreshold`   | Minimum consecutive successes for the probe to be considered successful after having failed (redis master pod) | `1`                                                  |
 | `master.readinessProbe.failureThreshold`   | Minimum consecutive failures for the probe to be considered failed after having succeeded.                     | `5`                                                  |
 | `master.priorityClassName`                 | Redis Master pod priorityClassName                                                                             | {}                                                   |
+| `volumePermissions.enabled`         | Enable init container that changes volume permissions in the registry (for cases where the default k8s `runAsUser` and `fsUser` values do not work)                                                               | `false`                                          |
```
Comment (desaintmartin) [inline stable/redis/README.md:170]: Isn't the value being "False" a breaking change?

Comment (javsalgar) [reply stable/redis/README.md:170]: Agree, bumping it

Comment (desaintmartin) [reply stable/redis/README.md:170]: Wouldn't it be better to keep backward compatibility and set it to true by default?

Comment (javsalgar) [reply stable/redis/README.md:170]: The issue with this is that `helm install redis` will not work in Openshift out of the box, and that's a User Experience issue that concerns me.

Comment (desaintmartin) [reply stable/redis/README.md:170]: Ah, so it does NOT break for regular k8s install by default? If so, okay!

Comment (javsalgar) [reply stable/redis/README.md:170]: Exactly, this was for a special case where the securityContext fsGroup was not enough

Comment (javsalgar) [reply stable/redis/README.md:170]: Anyways, it that would be the case, the `pull-charts-e2e` job would fail.

**Comment by desaintmartin** [comment]
Yes: document how to upgrade (i.e do nothing in 90% of cases :) )

**Comment by desaintmartin** [comment]
Thanks!
/ok-to-test

**Pull request merged**